src/cpp/lib/QtWidgets/QMessageBox/qmessagebox_wrap.cpp

```diff
@@ -0,0 +1,132 @@
+#include "QtWidgets/QMessageBox/qmessagebox_wrap.h"
+
+#include <QWidget>
+
+#include "Extras/Utils/nutils.h"
+#include "QtWidgets/QWidget/qwidget_wrap.h"
+
+Napi::FunctionReference QMessageBoxWrap::constructor;
+
+Napi::Object QMessageBoxWrap::init(Napi::Env env, Napi::Object exports) {
+  Napi::HandleScope scope(env);
+  char CLASSNAME[] = "QMessageBox";
+  Napi::Function func = DefineClass(
+      env, CLASSNAME,
+      {InstanceMethod("exec", &QMessageBoxWrap::exec),
+       InstanceMethod("setDefaultButton", &QMessageBoxWrap::setDefaultButton),
+       InstanceMethod("addButton", &QMessageBoxWrap::addButton),
+       StaticMethod("about", &StaticQMessageBoxWrapMethods::about),
+       StaticMethod("aboutQt", &StaticQMessageBoxWrapMethods::aboutQt),
+       QWIDGET_WRAPPED_METHODS_EXPORT_DEFINE(QMessageBoxWrap)});
+  constructor = Napi::Persistent(func);
+  exports.Set(CLASSNAME, func);
+  return exports;
+}
+
+NMessageBox* QMessageBoxWrap::getInternalInstance() { return this->instance; }
+QMessageBoxWrap::~QMessageBoxWrap() { extrautils::safeDelete(this->instance); }
+
+QMessageBoxWrap::QMessageBoxWrap(const Napi::CallbackInfo& info)
+    : Napi::ObjectWrap<QMessageBoxWrap>(info) {
+  Napi::Env env = info.Env();
+  Napi::HandleScope scope(env);
+  if (info.Length() == 1) {
+    Napi::Object parentObject = info[0].As<Napi::Object>();
+    QWidgetWrap* parentWidgetWrap =
+        Napi::ObjectWrap<QWidgetWrap>::Unwrap(parentObject);
+    this->instance = new NMessageBox(parentWidgetWrap->getInternalInstance());
+  } else if (info.Length() == 0) {
+    this->instance = new NMessageBox();
+  } else {
+    Napi::TypeError::New(env, "Wrong number of arguments")
+        .ThrowAsJavaScriptException();
+  }
+  this->rawData = extrautils::configureQWidget(
+      this->getInternalInstance(), this->getInternalInstance()->getFlexNode(),
+      false);
+}
+Napi::Value QMessageBoxWrap::exec(const Napi::CallbackInfo& info) {
+  Napi::Env env = info.Env();
+  Napi::HandleScope scope(env);
+
+  auto ret = this->instance->exec();
+  return Napi::Number::New(env, ret);
+}
+
+Napi::Value QMessageBoxWrap::setDefaultButton(const Napi::CallbackInfo& info) {
+  Napi::Env env = info.Env();
+  Napi::HandleScope scope(env);
+  if (info[0].IsNumber()) {
+    int button = info[0].As<Napi::Number>().Int32Value();
+    this->instance->setDefaultButton(QMessageBox::StandardButton(button));
+  } else {
+    Napi::Object buttonObject = info[0].As<Napi::Object>();
+    QPushButtonWrap* buttonWrap =
+        Napi::ObjectWrap<QPushButtonWrap>::Unwrap(buttonObject);
+    this->instance->setDefaultButton(buttonWrap->getInternalInstance());
+  }
+  return env.Null();
+}
+Napi::Value QMessageBoxWrap::addButton(const Napi::CallbackInfo& info) {
+  Napi::Env env = info.Env();
+  Napi::HandleScope scope(env);
+  if (info[0].IsNumber()) {
+    int button = info[0].As<Napi::Number>().Int32Value();
+    auto ret = this->instance->addButton(QMessageBox::StandardButton(button));
```

Comment:
This can crash the app if we try to attach a signal when we retrieve the button back since the button will be created by qt internally and wont be a npushbutton

Reply:
Any suggestions to fix that?

Reply:
Yep so we disable the use of this method from qt. We only allow addButton from QPushButtonWrap.